protocol/message.go

```diff
@@ -41,55 +52,180 @@ func (m *Message) DecodeData(keys map[string]string) error {
 	return nil
 }
 
-func (m *Message) Decrypt(cipherStr string, keys map[string]string) error {
+// EncodeData resets the current Encoding field to an empty string and starts
+// encoding data following the given `encoding` parameter.
+// `encoding` contains slash (/) separated values that EncodeData will read
+// from left to right to encode the current Data string.
+// To encode data using aes, the keys parameter must be present. Otherwise,
+// EncodeData will return an error.
+func (m *Message) EncodeData(encoding string, keys map[string]string) error {
```

Comment:
since `(*Message).Encrypt` requires fixed number of args - the `key` and vector `iv` shouldn't the signature be:

```go
func (m *Message) EncodeData(encoding string, key, iv []byte) error
```